### ISSUE TITLE
Linking static library failing with older Mingw64 ~gcc7.1 builds due to liberal use of experimental libraries

### ISSUE THREAD

**Issue by tilkinsc** [KhronosGroup/OpenXR-SDK-Source#236]
```
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: ./libopenxr_loader.a(filesystem_utils.cpp.obj):filesystem_utils.cpp:(.text$_Z28FileSysUtilsGetCanonicalPathRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERS4_+0x59): undefined reference to `__imp_PathCchCanonicalize'
collect2.exe: error: ld returned 1 exit status
```

The repo was built as a static library via,
```
mkdir build && cd build
cmake -G "MinGW Makefiles" ..
make all
```

Please migrate away from using `PathCchCanonicalize` located in static library `Pathcch.lib` as it is/was experimental. (build fail only shows when you try to link a program with it, as PathCchCanonicalize may be missing)

**Comment by aeikum** [comment]
See also #214.

This [library will be included in mingw-w64 v9](https://sourceforge.net/p/mingw-w64/mingw-w64/ci/6bd71e1d92586d3bff75aec353ca8b780e1c5cf9/), and I confirmed the loader will build with it. But that hasn't been released yet, and it won't help older mingw-w64 users anyway.

**Comment by rpavlik** [comment]
I'm pretty sure there's another issue about this too. I think we had agreed that you can actually no-op out the function using that (think there's only one, actually, so not that liberal 🤣 ): it's for following symlinks for manifests, which is a pattern on Linux and I asked for a Windows implementation.

Weird that it only shows up when you try to link a program with it - I had tried to get all the flags to force "eager" linking to find these issues early, though maybe that only works with dynamic linking. You probably want dynamic linking anyway, actually, in many cases, if nothing else to reduce the number of symbols exposed - if the default is still static on Windows that's just because we conflated that and having it system-wide-installed on Windows early on.

Looks like https://github.com/KhronosGroup/OpenXR-SDK-Source/pull/198 is what brought this usage in,

**Comment by rpavlik** [comment]
Fixed by #239.

**Comment by tilkinsc** [comment]
Excellent. I will be testing this later.
